Vizzuality/trase
# Performance/dashboards companies

## Pivotal Tracker

https://www.pivotaltracker.com/story/show/169391999

## Description

- in order to filter results by other nodes co-occurring on path, we use a subquery that runs on the flows table. to speed it up, in addition to filtering by path it also filters by context and years, same as main query
- that filtering has now been applied to all nodes (sources, destinations, exporters, importers), so that code is simpler
- if no selected co-occurring nodes specified, do not run the subquery at all
- there was a filter on co-occurring nodes in the main query, which used the `nodes_ids` computed array from dashboards_nodes tables. that is obsolete with the subquery filter in place, so was removed
- to save a bit of space, the `nodes_ids` precomputed column was dropped as this was the only place where it was used

## Testing instructions

go to dashboards and try different filters. should return the same nodes as in production, but faster.
